security/botan/Portfile

```diff
@@ -66,13 +74,13 @@ if {[string match *clang* ${configure.compiler}]} {
 if {${os.major} < 16} {
     # getentropy() appeared in OSX 10.12
     configure.args-append --without-os-features=getentropy
+    # clock_gettime supported by mac only from 10.12
+    configure.args-append --without-os-features=clock_gettime
```

Comment:
instead of disabling these option you might be able to use the `legacysupport` PortGroup which likely has added these functions (at least the `clock_gettime` is in there I think). I am not really sure what the effect of disabling this is and whether it's worthwhile looking into this...?

Reply:
legacysupport Portgroup was already in the Portfile this way:
``` 
PortGroup           legacysupport 1.0
# for arc4random_buf(), which is missing on 10.6 (Darwin 10)
legacysupport.newest_darwin_requires_legacy 10
```

and without `configure.args-append --without-os-features=clock_gettime` compilation fails:
```
src/lib/utils/os_utils.cpp:260:10: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
   const clockid_t clock_types[] = {
         ^~~~~~~~~
         clock_t
/usr/include/sys/_types/_clock_t.h:30:33: note: 'clock_t' declared here
typedef __darwin_clock_t        clock_t;
                                ^
src/lib/utils/os_utils.cpp:260:36: warning: zero size arrays are an extension [-Wzero-length-array]
   const clockid_t clock_types[] = {
                                   ^
src/lib/utils/os_utils.cpp:278:8: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
   for(clockid_t clock : clock_types)
       ^~~~~~~~~
       clock_t
/usr/include/sys/_types/_clock_t.h:30:33: note: 'clock_t' declared here
typedef __darwin_clock_t        clock_t;
                                ^
src/lib/utils/os_utils.cpp:281:12: error: no member named 'clock_gettime' in the global namespace
      if(::clock_gettime(clock, &ts) == 0)
         ~~^
src/lib/utils/os_utils.cpp:297:9: error: no member named 'clock_gettime' in the global namespace
   if(::clock_gettime(CLOCK_REALTIME, &ts) == 0)
      ~~^
src/lib/utils/os_utils.cpp:297:23: error: use of undeclared identifier 'CLOCK_REALTIME'
   if(::clock_gettime(CLOCK_REALTIME, &ts) == 0)
                      ^
1 warning and 5 errors generated.
```

botan checks if clock_getime is there, if it is, it wants to use `clockid_t`, if it is not, it uses something else (I don't remember what) that works on 10.11 (and probably the other old macos versions)

Same for `configure.args-append --without-os-features=getentropy`
```
In file included from src/lib/entropy/getentropy/getentropy.cpp:13:
/usr/include/sys/random.h:37:32: error: unknown type name 'u_int'
void read_random(void* buffer, u_int numBytes);
                               ^
/usr/include/sys/random.h:38:33: error: unknown type name 'u_int'
void read_frandom(void* buffer, u_int numBytes);
                                ^
/usr/include/sys/random.h:39:33: error: unknown type name 'u_int'
int  write_random(void* buffer, u_int numBytes);
                                ^
src/lib/entropy/getentropy/getentropy.cpp:27:9: error: no member named 'getentropy' in the global namespace
   if(::getentropy(buf.data(), buf.size()) == 0)
      ~~^
4 errors generated.
make: *** [build/obj/lib/entropy_getentropy.o] Error 1

```

How am I supposed to use `legacysupport`? I don't understand what is inside the portgroup and setting version to 1.0 or 1.1 doesn't change the problem. So these `--without-os-features=clock_gettime` & `--without-os-features=getentropy` seems necessary

Reply:
If I set `PortGroup           legacysupport 1.0` and remove `legacysupport.newest_darwin_requires_legacy 10` then it builds on 10.11 without setting `--without-os-features=getentropy` & `--without-os-features=clock_gettime`

(I had to put an additional patch to include sys/types.h to have u_int)

Not sure however what the impact will be on osx 10.6 (as per this comment: 
```
# for arc4random_buf(), which is missing on 10.6 (Darwin 10)
legacysupport.newest_darwin_requires_legacy 10
``` 

@reneeotten, Would you advise to proceed this way?

Reply:
I committed the change as a new commit 3727d92 to see it. (will squash later if needed)

Reply:
> If I set `PortGroup legacysupport 1.0` **and remove `legacysupport.newest_darwin_requires_legacy 10`** then it builds on 10.11 without setting `--without-os-features=getentropy` & `--without-os-features=clock_gettime`

well, you need to figure out what the *newest& system is that requires something from the `legacy_support` and put that number instead of the `10` in `legacysupport.newest_darwin_requires_legacy 10`. For example, if this is needed on El Capitan (which is `darwin 15`), the number should be `15`; you can find the overview [here](https://trac.macports.org/wiki/XcodeVersionInfo). 

Also, you should probably use the latest version of the PortGroup, I haven't looked what the difference between the two is, but presumably the newer the better ;)

Reply:
Sorry, I meant PortGroup legacysupport 1.1 in my previous comment. So it already is the latest.

concerning the version of `legacysupport.newest_darwin_requires_legacy` I set it to 15 since botan needs clock_gettime & getentropy which are missing until osx 10.11 (=darwin 15)

Committed in e76ed73

Reply:
I finally squashed the commits.